File: rowanlee92/graph/FloodFill.java
package com.janjanee.algo.graph;

public class FloodFill {

    public static int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        int start = image[sr][sc];

        if (start != newColor)
            dfs(image, sr, sc, start, newColor);

        return image;
    }

    private static void dfs(int[][] image, int sr, int sc, int start, int newColor) {
        if (image[sr][sc] == start) {
            image[sr][sc] = newColor;
            if (sc >= 1) dfs(image, sr, sc - 1, start, newColor);
            if (sr >= 1) dfs(image, sr - 1, sc, start, newColor);
            if (sc < image[0].length - 1) dfs(image, sr, sc + 1, start, newColor);
            if (sr < image.length - 1) dfs(image, sr + 1, sc, start, newColor);
        }
    }

}

File: yeseulo/graph/floodFill.js
/**
 * Leetcode - 733. Flood Fill
 * @param {number[][]} image
 * @param {number} sr
 * @param {number} sc
 * @param {number} newColor
 * @return {number[][]}
 */
var floodFill = function (image, sr, sc, newColor) {
  let start = image[sr][sc];
  if (newColor === start) return image;

  const dfs = (img, row, col) => {
    if (
      row < 0 ||
      col < 0 ||
      row >= img.length ||
      col >= img[0].length ||
      img[row][col] !== start
    )
      return;
    img[row][col] = newColor;

    dfs(img, row - 1, col);
    dfs(img, row + 1, col);
    dfs(img, row, col - 1);
    dfs(img, row, col + 1);

    return img;
  };

  return dfs(image, sr, sc);
};